src/containers/User/UserRegister.tsx

```diff
@@ -12,7 +12,7 @@ const UserRegister: React.FC = () => {
     .then((loginResult) => {
       console.log('Result', loginResult)
       localStorage.setItem('persistJWT', loginResult.JWT)
-      localStorage.setItem('username', loginResult.username)
+      localStorage.setItem('user', JSON.stringify(loginResult.user))
```

Comment:
Do we need to keep the current user in place, in case we reload the page? But in this case - after reloading I can't see where it is updating the context to use the current user in the evaluation

Reply:
> But in this case - after reloading I can't see where it is updating the context to use the current user in the evaluation

The UserArea get "user" and "templatePermissions" from `useGetUserPermissions`, which it then sets to context (near the top of UserArea component)

> Do we need to keep the current user in place, in case we reload the page?

I was doing it so the "user" data could persist between login and loading `useGetUserPermissions`, but now that I think about it, we could save the userdata in Context in the login component and then it'd be already available everywhere else.

Although we still need to persist in in localStorage so that the user data is available next visit/reload.

Do you think it's a problem?

Reply:
Okay, I've made the changes I suggested above. I think it's clearer now and keeps the user info/context separate from the Permissions fetching.

So, in summary:
- After successful login, the Login component writes the user info to Context and localStorage
- When reloading, and "loginCheck" confirms that user is already logged in (i.e. has valid JWT and user data), the UserArea fetches the user data from localStorage and stores it in context (it only does this if it can't find a currentUser already in context).

Reply:
I understood now. That's what I was thinking, if the user already comes from Graphql query and that runs on page reaload why the need to store it locally, but I see that the information could delay to display. I think this is good.